### ISSUE TITLE
Better CMake stuff for finding dbus

### ISSUE THREAD

**Issue by tammojan** [radio-astro/casasynthesis#3]
If we cannot get rid of the dbus dependencies, at least the CMake detection should be good. I ran into a case (ubuntu 14.04) where cmake found dbus, but at build time dbus.h was not found. I'll investigate. Perhaps we can borrow a FindDBus.cmake from another project.

**Comment by o-smirnov** [comment]
I say we put in the extra effort to get rid of dbus. It's evil. Where is it
linked in, Gijs?

On Sat, Sep 19, 2015 at 1:46 PM, Tammo Jan Dijkema <notifications@github.com

**Comment by gijzelaerr** [comment]
it is used quite deep in the code on various places.

2015-09-19 13:48 GMT+02:00 Oleg Smirnov notifications@github.com:

> I say we put in the extra effort to get rid of dbus. It's evil. Where is it
> linked in, Gijs?
> 
> On Sat, Sep 19, 2015 at 1:46 PM, Tammo Jan Dijkema <
> notifications@github.com
> 
> > wrote:
> > 
> > If we cannot get rid of the dbus dependencies, at least the CMake
> > detection should be good. I ran into a case (ubuntu 14.04) where cmake
> > found dbus, but at build time dbus.h was not found. I'll investigate.
> > Perhaps we can borrow a FindDBus.cmake from another project.
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/radio-astro/casasynthesis/issues/3.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/radio-astro/casasynthesis/issues/3#issuecomment-141654062
> .

## 

Gijs Molenaar
http://pythonic.nl

**Comment by o-smirnov** [comment]
I think we need to sit down together and go over it...

**Comment by tammojan** [comment]
I agree that getting rid of it is cleaner, so we shouldn't spend major effort on the cmake stuff.

For getting rid of it, we could be inspired by how this was done for casacore/casarest (where @gervandiepen probably did the same).

**Comment by o-smirnov** [comment]
@gijzelaerr do you have some quick magicky tool for showing all the code that invokes dbus?

**Comment by gijzelaerr** [comment]
remove the DBUS cmake finding logic and linking stuff, remove headers and
try compile.

Or use something like CLion to do static code analysis.

2015-09-19 13:56 GMT+02:00 Oleg Smirnov notifications@github.com:

> @gijzelaerr https://github.com/gijzelaerr do you have some quick
> magicky tool for showing all the code that invokes dbus?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/radio-astro/casasynthesis/issues/3#issuecomment-141655099
> .

## 

Gijs Molenaar
http://pythonic.nl

**Comment by tammojan** [comment]
Easier: look at the old Casarest code, grep for CASA_STANDALONE. That is the ifdef thing that Ger used to remove all the dbus (and probably pgplot) stuff. We simply need to put those ifdefs in Casasynthesis as well.

**Comment by o-smirnov** [comment]
I think the problem is @gijzelaerr is not familiar with the synthesis code at all, so it's hard for him to just ifdef stuff out blindly. This is why me and/or @gervandiepen should look at it with him.
